crates/notion-core/src/serial/package.rs

```diff
@@ -0,0 +1,117 @@
+extern crate serde;
+extern crate serde_json;
+
+use package_info::PackageInfo;
+
+use serde::de::{Deserialize, Deserializer, Error, MapAccess, Visitor};
+
+use std::collections::HashMap;
+use std::fmt;
+use std::hash::Hash;
+use std::marker::PhantomData;
+use std::ops::{Deref, DerefMut};
+
+// wrapper for HashMap to use with deserialization
+pub struct BinMap<K, V>(HashMap<K, V>);
+
+impl<K, V> Deref for BinMap<K, V> {
+    type Target = HashMap<K, V>;
+
+    fn deref(&self) -> &HashMap<K, V> {
+        &self.0
+    }
+}
+
+impl<K, V> DerefMut for BinMap<K, V> {
+    fn deref_mut(&mut self) -> &mut HashMap<K, V> {
+        &mut self.0
+    }
+}
+
+#[derive(Deserialize)]
+pub struct Info {
+    pub name: String,
+    // the "bin" field can be a map or a string
+    // (see https://docs.npmjs.com/files/package.json#bin)
+    #[serde(default)] // handles Option
+    pub bin: Option<BinMap<String, String>>,
+}
+
+impl Info {
+    pub fn into_package_info(self) -> PackageInfo {
+        let mut map = HashMap::new();
+        if self.bin.is_some() {
+            for (name, path) in self.bin.unwrap().iter() {
+                if name == "" {
```

Comment:
This is because down in the serializer, for the `visit_str` function, I'm using a placeholder of `""` for the package name (since it is not guaranteed to be known at that point). And then here I'm just replacing that with the actual package name now that it is available.

Doing that so that the returned map will always contain mapping from the binary name to the path, instead of having just the path.

Reply:
I just added a comment about this in the code

Reply:
Ah, I get it now. Thanks for adding the comments!